pkg/cable/wireguard/driver.go

```diff
@@ -370,6 +402,27 @@ func discoverInternalIP(cidrs []net.IPNet) (string, error) {
 	return "", nil
 }
 
+// look at local subnets, cluster cidrs, and vxlan subnets
+func (w *wireguard) discoverInternalIP(localSubnets []string, localCluster types.SubmarinerCluster) (string, error) {
+	cidrs := parseSubnets(localSubnets)
+	ip, err := matchLocalIP(cidrs)
```

Comment:
Can we use the discoverCNIInterface API here by moving it to utils?

Reply:
+1 for having `discoverCNIInterface` in utils. This function is a bit different -- it looks first for a `localSubnet` CIDR and then for cluster CIDR. It uses `matchLocalIP()` which is perhaps more similar. 
As we discussed with @mangelajo, longer term, the local tunnel address (both GN and non-GN cases) should ideally be decided before the cable driver is started and pass it as parameter. This would allow all iptable rules to use the address (eg for GN SNAT). Also, we should consider using `ip route get <some cluster IP>` to figure out the source address used by CNI.

Reply:
Yup. We can do any such changes in a separate PR.